### PR TITLE
SOLR-15337 Avoid XPath in solrconfig.xml parsing 

### PR DESCRIPTION
**Pull request by noblepaul**
See https://issues.apache.org/jira/browse/SOLR-15337 for details

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/core/SolrConfig.java [hunk 1]
```diff
@@ -170,139 +201,178 @@ public static SolrConfig readFromResourceLoader(SolrResourceLoader loader, Strin
...
+            log.debug("LOADED_FROM_CACHE");
+            root = cfg.data;
+          } else {
+            readXml(loader, name, configCache, rp);
+          }
+        }
+      }
+    }
+    if(root == null) {
+      readXml(loader, name, configCache,new ResourceProvider(loader.openResource(name)) );
```
Comment (hiteshk25) [inline solr/core/src/java/org/apache/solr/core/SolrConfig.java:226]: are we calling readXml(..) again here, if two calls happens simultaneously for same config?

Comment (noblepaul) [reply solr/core/src/java/org/apache/solr/core/SolrConfig.java:226]: that's if it's not read already

**Comment by hiteshk25** [comment]
I ran the unit tests on this branch and those passed

**Review by chatman — APPROVED** [review]
Tested using rolling restart tests, it is faster with this test (38s vs 44s for 640 cores/node).

**Comment by hiteshk25** [comment]
LGTM

**Pull request merged**